### PR TITLE
Add blog articles and display them

### LINKED ISSUE #175: Hosting blog posts & content on our website
the idea is to have posts visible on https://usefusion.app/blog

We'd want to be able to update the entries here on the fly. 

Here's a guide to adding a markdown blog to a nextjs site. I think we can just follow this https://dev.to/ethand91/creating-a-markdown-blog-with-nextjs-1dci

[Link to current posts on Substack we want to move](https://neurofusion.substack.com)

Eventually, this'll become how we host other forms of dynamic posts / updates for Fusion

### PR DESCRIPTION
**Pull request by peoray**
Resolves #175

### CODE REVIEW

File: neurofusion/next-client/src/utils/blog.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    markdownBody: content,
+  };
+}
+
+export async function getAllPostsWithFrontMatter() {
+  const files = fs.readdirSync(path.join(root, "src/articles"));
+
+  // @ts-ignore
+  return files.reduce((allPosts, postSlug) => {
+    const source = fs.readFileSync(path.join(root, "src/articles", postSlug), "utf8");
```
Comment (oreHGA) [inline neurofusion/next-client/src/utils/blog.ts:28]: so this won't work after the project builds & deploy

**Comment by oreHGA** [comment]
I think we can split this PR into smaller chunks to reduce how much you're having to do at once?
- framework for the blog posts (which I think is mostly done, pending my comment about the folder for the data)
- the actual blog post entries formatted (this can come in afterwards)

**Comment by peoray** [comment]
@oreHGA that won't be necessary, just gimme till this weekend to update the current pr. Already started adding the images and styling

**Comment by peoray** [comment]
@oreHGA that won't be necessary, just gimme till this weekend to update the current pr. Already started adding the images and styling

File: neurofusion/next-client/src/utils/blog.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    const { data } = matter(source);
+
+    return [
+      {
+        frontMatter: data,
+        slug: postSlug.replace(".md", ""),
+      },
+      ...allPosts,
```
Comment (oreHGA) [inline neurofusion/next-client/src/utils/blog.ts:36]: let's have the slug as part of the frontMatter in the markdown file

File: neurofusion/next-client/src/utils/blog.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+  // @ts-ignore
+  return files.reduce((allPosts, postSlug) => {
+    const source = fs.readFileSync(path.join(root, "src/articles", postSlug), "utf8");
+    const { data } = matter(source);
+
+    return [
+      {
+        frontMatter: data,
+        slug: postSlug.replace(".md", ""),
```
Comment (oreHGA) [inline neurofusion/next-client/src/utils/blog.ts:34]: let's have the slug as part of the frontMatter in the markdown file

File: neurofusion/next-client/src/components/ui/card/blog-card.tsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+              return new Date(b.frontMatter.publishedDate).getTime() - new Date(a.frontMatter.publishedDate).getTime();
+            })
+            .map((post) => {
+              return (
+                <Link href={{ pathname: `/blog/${post.slug}` }} key={post.slug}>
+                  <article key={post.slug} className="flex justify-between p-2">
+                    <div className="flex flex-col mr-8 gap-y-4">
+                      <h2>{post.frontMatter.title}</h2>
+                      <p>{post.frontMatter.description}</p>
+                      <p>{formatDate(post.frontMatter.publishedDate)}</p>
```
Comment (oreHGA) [inline neurofusion/next-client/src/components/ui/card/blog-card.tsx:22]: dayjs has inbuilt methods we use across the project and we can use for this https://day.js.org/docs/en/parse/string-format

**Comment by oreHGA** [comment]
Hey @pe

> @oreHGA that won't be necessary, just gimme till this weekend to update the current pr. Already started adding the images and styling

Hey @peoray , wanted to check in on this. Are you able to push what you have? (doesn't need to be perfect?)

I'm making more updates to site this week. If I don't hear more from you I'd assume you're busy with other stuff (& that's okay) and I'll take this over

File: neurofusion/next-client/src/components/ui/card/blog-card.tsx (hunk 4 above)
Comment (peoray) [reply neurofusion/next-client/src/components/ui/card/blog-card.tsx:22]: there's a date formatter method already

File: neurofusion/next-client/src/utils/blog.ts (hunk 3 above)
Comment (peoray) [reply neurofusion/next-client/src/utils/blog.ts:34]: atm, the slug is the name of the file which I don't think I did a good job of :)

File: neurofusion/next-client/src/components/ui/card/blog-card.tsx (hunk 4 above)
Comment (peoray) [reply neurofusion/next-client/src/components/ui/card/blog-card.tsx]: used dayjs instead then

File: neurofusion/next-client/src/utils/blog.ts (hunk 3 above)
Comment (oreHGA) [reply neurofusion/next-client/src/utils/blog.ts:34]: thanks, I'll make the update in the next PR

**Comment by oreHGA** [comment]
Congrats on your first PR @peoray! I'll tag you when I make the changes for the blog slugs

**Comment by oreHGA** [comment]
Made the changes for the blog slugs in #184

**Pull request merged**